src/test/java/de/tum/in/www1/artemis/ConsistencyCheckBitbucketBambooIntegrationTest.java

```diff
@@ -0,0 +1,70 @@
+package de.tum.in.www1.artemis;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.test.context.support.WithMockUser;
+
+import de.tum.in.www1.artemis.service.ConsistencyCheckServiceTest;
+
+public class ConsistencyCheckBitbucketBambooIntegrationTest extends AbstractSpringIntegrationBambooBitbucketJiraTest {
+
+    @Autowired
+    private ConsistencyCheckServiceTest consistencyCheckServiceTest;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        consistencyCheckServiceTest.setup(this);
+        bambooRequestMockProvider.enableMockingOfRequests();
+        bitbucketRequestMockProvider.enableMockingOfRequests();
+    }
+
+    @AfterEach
+    public void tearDown() throws Exception {
+        bitbucketRequestMockProvider.reset();
+        bambooRequestMockProvider.reset();
+        database.resetDatabase();
+    }
+
+    /**
+     * Test consistencyCheck feature with programming exercise without
+     * inconsistencies
+     * @throws Exception if an error occurs
+     */
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void checkConsistencyOfProgrammingExercise_noErrors() throws Exception {
+        consistencyCheckServiceTest.checkConsistencyOfProgrammingExercise_noErrors();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void checkConsistencyOfProgrammingExercise_missingVCSProject() throws Exception {
+        consistencyCheckServiceTest.checkConsistencyOfProgrammingExercise_missingVCSProject();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void checkConsistencyOfProgrammingExercise_missingVCSRepos() throws Exception {
+        consistencyCheckServiceTest.checkConsistencyOfProgrammingExercise_missingVCSRepos();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void checkConsistencyOfProgrammingExercise_buildPlansMissing() throws Exception {
+        consistencyCheckServiceTest.checkConsistencyOfProgrammingExercise_buildPlansMissing();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void checkConsistencyOfProgrammingExercise_isLocalSimulation() throws Exception {
+        consistencyCheckServiceTest.checkConsistencyOfProgrammingExercise_isLocalSimulation();
+    }
+
+    @Test
+    @WithMockUser(username = "instructor1", roles = "INSTRUCTOR")
+    public void checkConsistencyOfCourse() throws Exception {
+        consistencyCheckServiceTest.checkConsistencyOfCourse();
+    }
```

Comment:
Maybe additional tests that make sure that the two endpoints return a 403 forbidden for non-instructors should be added.

Reply:
Two additional tests for unauthorized cases have been added :+1

Reply:
I originally intended the test to be run with a regular student in the course as user. Using any instructor that isn’t one for the current course woks as well. :+1: